src/pkg/mcp/setup.go

```diff
@@ -298,18 +306,23 @@ func handleStandardConfig(configPath string) error {
 			return fmt.Errorf("failed to read config file: %w", err)
 		}
 
-		// Parse the JSON
-		if err := json.Unmarshal(data, &config); err != nil {
-			// If we can't parse it, start fresh
-			config = MCPConfig{
-				MCPServers: make(map[string]MCPServerConfig),
+		// Parse the JSON into a generic map to preserve all settings
+		if err := json.Unmarshal(data, &existingData); err != nil {
+			return fmt.Errorf("failed to unmarshal existing config: %w", err)
+		}
+
+		// Try to extract MCPServers from existing data
+		if mcpServersData, ok := existingData["mcpServers"]; ok {
+			// Convert back to MCPConfig structure
+			mcpServersJSON, err := json.Marshal(map[string]any{"mcpServers": mcpServersData})
+			if err != nil {
+				return fmt.Errorf("failed to marshal mcpServers: %w", err)
 			}
+			json.Unmarshal(mcpServersJSON, &config)
```

Comment:
You are not handling a potential error here.
```suggestion
			err = json.Unmarshal(mcpServersJSON, &config)
			if err != nil {
				return fmt.Errorf("failed to unmarshal mcpServers: %w", err)
			}
```